backend/src/main/java/ubb/courses/backend/models/Media.java

```diff
@@ -0,0 +1,24 @@
+package ubb.courses.backend.models;
+
+import lombok.*;
+import ubb.courses.backend.models.MediaTypes.MediaTypeEnum;
+import ubb.courses.backend.models.generic.BaseEntity;
+
+import javax.persistence.*;
+
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "media_type", discriminatorType = DiscriminatorType.INTEGER)
```

Comment:
For better readability it would be bettern if the disciminator type will be string and it will contain the value from the enum such that we won't be checking back and forth which is which as a type

Reply:
done